department-of-veterans-affairs/vets-api
# SPOOL: Track spool file creation by RPO

## Description of change


All changes are in support of preventing multiple files for a single RPO per day being created in CreateDailySpoolFiles

The goal is to create this flow for when a failure happens during a scheduled run:
- failure for eastern RPO happens, only western RPO is spool file is created
- retry attempt 1 out of 5: doesn’t make a new western RPO cause not a “new” run of job, successfully makes an eastern RPO file
- sftps western RPO from first run and eastern RPO from first retry

## Original issue(s)
department-of-veterans-affairs/va.gov-team#21243
department-of-veterans-affairs/va.gov-team#21246

## Things to know about this PR

### Testing

1. Submit a form for a RPO
2. Check `spool_file_events` table notice no rows
3. Run CreateDailySpoolFiles job
4. In `spool_file_events` table notice there is now a row for each RPO the form submissions would go to, this row should have a `successful_at` timestamp and a `retry_attempt` of 0
5. In `vets-api/tmp` there should be a file with format `<rpo>_<timestamp>_vetsgov.spl` with first submission
6. Run CreateDailySpoolFiles job
7. Notice no updates were made to either `spool_file_events` or `<rpo>_<timestamp>_vetsgov.spl`
8. Remove `successful_at` timestamp from the row
9. Delete the file with format same RPO and `monthdayyear` values as created in step 5
10. Submit a form for the same RPO again
11. Run CreateDailySpoolFiles job
12. The file with format `<rpo>_<timestamp>_vetsgov.spl` should now contain the new submission
13. In `spool_file_events` table the row for the RPO should have a `successful_at` timestamp and a `retry_attempt` of 1
